File: isAnagram.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

function isAnagram() {
  // with user prompt
  rl.question("Enter a string and Press Enter: ", (str1) => {
    let sortedStr1 = str1.toLowerCase().split("").sort().join("");
    rl.question("Now enter another string: ", (str2) => {
      let sortedStr2 = str2.toLowerCase().split("").sort().join("");
      const result = sortedStr1 === sortedStr2;
      try {
        console.log(`Is anagram? ${result}`);
      } catch (error) {
        // this is for the test case "display an error message if the function could not complete."
        throw new Error("Error");
      }
      rl.close();
    });
  });
}

isAnagram();

// without user prompt
// function isAnagram(str1, str2) {
//   // convert the strings to lower case for the same and Seam edge-case.
//   // separating each letter as an individual letter in an array with split("")
//   // sorting them alphabetically with .sort()
//   // combine them again back into a string from an array with .join('')
//   let sortedStr1 = str1.toLowerCase().split("").sort().join("");
//   let sortedStr2 = str2.toLowerCase().split("").sort().join("");
//   return sortedStr1 === sortedStr2; // this will return true or false based on if the 2 strings given are anagrams.
// }
// console.log(isAnagram("same", "Seam")); // => true
// console.log(isAnagram("hare", "hear")); // => true
// console.log(isAnagram("cat", "dog")); // => false

File: is_anagram.rb
def is_anagram
  
  def prompt(*args)
      print(*args)
      gets.chomp
  end
  
  answer1 = prompt "Please enter the first string: "

  sorted_str1 = answer1.downcase.split('').sort().join('')

  answer2 = prompt "Please enter the second string: "

  sorted_str2 = answer2.downcase.split('').sort.join('')

  result = sorted_str1 === sorted_str2 # will return true or false.


  begin

  puts "are #{answer1} and #{answer2} anagrams? #{result ? 'Yes' : 'No'}" # if true it will print yes, else it will print no

  rescue RuntimeError # catch error

  puts "Runtime error encountered and rescued."
  end

end

is_anagram
